src/auth.rs

```diff
@@ -133,52 +201,92 @@ impl<'r> FromRequest<'r> for CreateAuthWrapper {
-                    };
-                    match create_orbit(
-                        *orbit_id,
-                        config.database.path.clone(),
-                        TezosBasicAuthorization {
-                            controllers: vec![vm],
-                        },
-                        &auth_data,
-                    )
-                    .await
-                    {
-                        Ok(Some(orbit)) => Outcome::Success(Self(orbit)),
-                        Ok(None) => {
+                };
+                let controllers = match &token {
+                    AuthTokens::Tezos(token_tz) => {
+                        match method {
+                            "tz" => {}
+                            _ => {
+                                return Outcome::Failure((
+                                    Status::BadRequest,
+                                    anyhow!("Incorrect Orbit ID"),
+                                ))
+                            }
+                        };
+                        if params.get("address") != Some(&token_tz.pkh.as_str()) {
                             return Outcome::Failure((
-                                Status::Conflict,
-                                anyhow!("Orbit already exists"),
-                            ))
-                        }
-                        Err(e) => Outcome::Failure((Status::InternalServerError, e)),
+                                Status::Unauthorized,
+                                anyhow!("Incorrect PKH param"),
+                            ));
+                        };
+                        let vm = DIDURL {
+                            did: format!("did:pkh:tz:{}", &token_tz.pkh),
+                            fragment: Some("TezosMethod2021".to_string()),
+                            ..Default::default()
+                        };
+                        vec![vm]
                     }
+                    AuthTokens::ZCAP(ZCAPTokens { invocation, .. }) => {
+                        let vm = match invocation.proof.as_ref().and_then(|p| {
+                            p.verification_method.as_ref().map(|v| DIDURL::from_str(&v))
+                        }) {
+                            Some(Ok(v)) => v,
+                            _ => {
+                                return Outcome::Failure((
+                                    Status::Unauthorized,
+                                    anyhow!("Invalid Delegation Verification Method"),
+                                ))
+                            }
+                        };
+                        vec![vm]
+                    }
+                    _ => {
+                        return Outcome::Failure((
+                            Status::Unauthorized,
+                            anyhow!("Missing Authorization"),
+                        ))
+                    }
+                };
+                match create_orbit(
```

Comment:
I'm confused about the relation between this create and the other `create_orbit` for the create endpoint

Reply:
the endpoint used `create_orbit` when an authz token is not present, relying on the allowlist to provide the right info. The two methods have now been separated